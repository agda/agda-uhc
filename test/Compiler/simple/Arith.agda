module Arith where

open import Common.IO
open import Common.Nat renaming (_âˆ¸_ to _-_)  -- workaround for #1855
open import Common.Unit

test : Nat
test = 4

foobar : Nat -> Nat
foobar zero = zero
foobar (suc n) = suc (suc n)

main : IO Unit
main =
--  n <- readNat ,
  printNat 0 ,,
  printNat (0 + 1) ,,
  printNat (1 * 2) ,,
  printNat (suc (suc (suc (suc zero))) - suc zero) ,,
  printNat test    ,,
  printNat (foobar 4) ,,
--  printNat n ,,
  return unit
