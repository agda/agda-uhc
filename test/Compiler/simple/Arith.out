EXECUTED_PROGRAM

ret > ExitSuccess
out > 012345
